File: Codes/Garnetwzy/160.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode *p1 = headA;
        ListNode *p2 = headB;
        int len1 = 0;
        int len2 = 0;
        while(p1) {
            len1++;
            p1 = p1->next;
        }
        while(p2) {
            len2++;
            p2 = p2->next;
        }
        p1 = headA;
        p2 = headB;
        int diff = abs(len1 - len2);
        if(len1 < len2) {
            while(diff > 0) {
                p2 = p2->next;
                diff--;
            }
        }else if(len1 > len2) {
            while(diff > 0) {
                p1 = p1->next;
                diff--;
            }
        }
        while(p1 != p2) {
            p1 = p1->next;
            p2 = p2->next;
        }
        return p1;
    }
};

File: codes/HellDream/160.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        int lenA = getLength(headA);
        int lenB = getLength(headB);
        ListNode p1 = headA, p2 = headB;
        if(lenA>lenB){
            for(int i=0;i<lenA-lenB;i++){
                p1 = p1.next;
            }
        }else{
            for(int i=0;i<lenB-lenA;i++){
                p2 = p2.next;
            }
        }
        while(p1!=null&&p2!=null){
            if(p1==p2) return p1;
            p1 = p1.next;
            p2 = p2.next;
        }
        return null;
    }

    public int getLength(ListNode node){
        int len = 0;
        while(node!=null){
            node  =node.next;
            len++;
        }
        return len;
    }
}